### PR TITLE
Task 3 - Add InventoryDAO.delete(String) method with tests

### PR DESCRIPTION
**Pull request by jditch**
See ticket #55 for specifications
This pull also adjusts previous InventoryDAO tests for readability purposes

### CODE REVIEW

File: starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java [hunk 1]
```diff
@@ -61,9 +63,44 @@ public void create() {
...
+    inventory.setName(NAME);
+    inventory.setProductType(PRODUCT_TYPE);
+    this.inventoryDAO.create(inventory);
+    String createdID = inventory.getId();
+    Optional<Inventory> deletedInventory = this.inventoryDAO.delete(createdID);
+    
+    Assert.assertTrue(deletedInventory.isPresent());
+    Assert.assertEquals(inventory.getName(), deletedInventory.get().getName());
+    Assert.assertEquals(inventory.getProductType(), deletedInventory.get().getProductType());
+    Assert.assertEquals(createdID, deletedInventory.get().getId());
```
Comment (lndyaouie) [inline starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java]: You can remove the above `Assert.assertEquals` statements for inventory `name`, `productType`, and `id` as we know the correct inventory was deleted if line 85 is true.

File: starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java [hunk 2]
```diff
@@ -61,9 +63,44 @@ public void create() {
...
+   * Test Delete method where the inventory to be deleted exists
+   */
+  @Test
+  public void deleteInventoryExists() {
+    Inventory inventory = new Inventory();
+    inventory.setName(NAME);
+    inventory.setProductType(PRODUCT_TYPE);
+    this.inventoryDAO.create(inventory);
+    String createdID = inventory.getId();
+    Optional<Inventory> deletedInventory = this.inventoryDAO.delete(createdID);
```
Comment (lndyaouie) [inline starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java]: `createdId` can be replaced with the getter here; remove the `createdId` variable afterwards.

File: starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java [hunk 3]
```diff
@@ -61,9 +63,44 @@ public void create() {
...
+  
+  /**
+   * Test Delete method where the inventory to be deleted exists
+   */
+  @Test
+  public void deleteInventoryExists() {
+    Inventory inventory = new Inventory();
+    inventory.setName(NAME);
+    inventory.setProductType(PRODUCT_TYPE);
+    this.inventoryDAO.create(inventory);
```
Comment (lndyaouie) [inline starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java]: Assign the created inventory to a new variable (e.g., `Inventory createdInventory = this.inventoryDAO.create(inventory);`).

After creating an inventory, add an assertion to check the Inventory Collection before removal like on line 89.

**Review by lndyaouie — CHANGES_REQUESTED** [review]

File: starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java (hunk 1 above)
Comment (jditch) [reply starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java]: Lines removed in commit f0f86e97102500321c70556b975a146356f674fd

File: starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java (hunk 3 above)
Comment (jditch) [reply starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java]: Changes made in commit f0f86e97102500321c70556b975a146356f674fd

File: starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java (hunk 2 above)
Comment (jditch) [reply starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java]: Changes made in commit f0f86e97102500321c70556b975a146356f674fd

**Pull request merged**